ui: jobs page

vladlos opened the pull request:
jobs page redesign

Resolves: #43911

Release note (ui): none

vladlos committed:
ui: jobs page

styled jobs page and table, added pagination

Release note (ui): none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/44681)
<!-- Reviewable:end -->

vladlos committed:
ui: job page

added job details page

Release note (ui): none

dhartunian commented on pkg/ui/src/views/jobs/index.tsx:
```diff
@@ -74,60 +73,68 @@ const showOptions = [
+    case JOB_STATUS_CANCELED:
+      return "Import failed";
+    default:
+      return status;
```
can you capitalize the string to match the other renamed statuses?

dhartunian commented on pkg/ui/src/views/jobs/jobDetails.tsx:
```diff
@@ -0,0 +1,200 @@
+    }
+  }
+
+  renderContent = () => {
```
Components with a lot of nested `.renderX()` calls are difficult to read and debug. Can you extract into separate components with props?

I would love to see component names clarify the render hierarchy instead of `this.renderBlah`.

For example, try simple presentational components like `<JobStatus job={job} />`. Then we can introduce much easier testing of the low level components in enzyme. Right now, the `JobDetails` component is very tricky to test because it requires too many inputs.

dhartunian replied:
Components will be extracted.

dhartunian commented:
I'll pick up this PR and get it merged. Won't be addressing follow-ups in #45259.

dhartunian committed:
ui: capitalize statuses when renaming for display

Release note: none

dhartunian committed:
ui: refactor; extract jobStatusDropdown

Release note: none

dhartunian committed:
ui: refactor; extract JobStatusCell and JobTable components

Moving towards a single-component-per-file design that's easier
to manage and modify

Release note: none

dhartunian committed:
ui: refactor; extract more job components

Duration, FractionCompleted, HighwaterTimestamp, Progress
components have been extracted from JobStatusCell

Release note: none

dhartunian committed:
ui: refactor; extract JobDescriptionCell component

Release note: none

dhartunian committed:
ui: refactor; inline JobStatusDropdown component

This extraction was unnecessary. Component is very simple and has no logic.

Release note: none

dhartunian committed:
ui: refactor; move renamedStatuses to separate file

Release note: none

dhartunian committed:
ui: refactor; move job status constants to separate file

Release note: none

dhartunian committed:
ui: refactor; use jobHasOneOfStatuses function

Release note: none

dhartunian committed:
ui: clean up classes in JobDescriptionCell

Repeated classname in hierarchy was unnecessary.
Removed usage in `<Link>`.

Release note: none

dhartunian committed:
ui: extracted Duration component must return null

When a duration cannot be rendered we must return null.
Previously this was an inline function that did not have such a restriction.

Now that we've extracted to a React component we need to explicitly
return null.

This is to preserve older functionality prior to refactor.

Release note: none

dhartunian committed:
ui: reset pagination when new jobs are loaded

This fixes the bug where paging through and then changing a
filter would render an empty list because the new filtered
list only had one page.

Now the `current` page state will reset to 1 any time a new list
of jobs is pushed to the table component.

Added an enzyme test to verify.

Release note: none

dhartunian committed:
ui: add license to new files

Release note: none

dhartunian committed:
ui: refactor; use Duration component in JobDetails

The component we extracted from the JobTable can now
be dropped right into JobDetails and replace an inline function.

Release note: none

dhartunian committed:
ui: refactor; use Progress component in JobDetails

Two props needed to be added to Progress to make it compatible
with the usage in JobDetails but otherwise they are identical

Release note: none

dhartunian commented:
bors r+
